pwnlib/util/cyclic.py

```diff
@@ -289,12 +280,8 @@ def cyclic_metasploit(length = None, sets = None):
     sets = sets or [ string.ascii_uppercase.encode(), string.ascii_lowercase.encode(), string.digits.encode() ]
     out = bytearray()
 
-    for ndx, c in enumerate(metasploit_pattern(sets)):
-        if length != None and ndx >= length:
-            break
-        else:
-            out.append(c)
-
+    generator = metasploit_pattern(sets)
+    out = [next(generator) for _ in range(length)] if length != None else list(generator)
```

Comment:
You could also use [iters.take](/Gallopsled/pwntools/blob/2116bf3118612a5f53a9a02a8d46ed2f8b1cefcb/pwnlib/util/iters.py#L72) here I guess, as it works also with `n=None`.

Reply:
Will do.